### PR TITLE
chore(broker): replace SocketAddress with InetSocketAddress

### LINKED ISSUE #3867: Broker will not start on host names longer then 128 bytes
**Describe the bug**

It seems that longer host names, like longer benchmark names make problems on broker startup.
https://github.com/zeebe-io/zeebe/blob/develop/transport/src/main/java/io/zeebe/transport/impl/SocketAddress.java#L24

**To Reproduce**

Just create a benchmark with a long name.

**Expected behavior**
No problem.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
java.lang.IllegalArgumentException: Host length exceeds max length (139 > 128 bytes)
at io.zeebe.transport.impl.SocketAddress.checkHostLength (SocketAddress.java:99)
at io.zeebe.transport.impl.SocketAddress.host (SocketAddress.java:63)
at io.zeebe.transport.impl.SocketAddress.host (SocketAddress.java:71)
at io.zeebe.transport.impl.SocketAddress.<init> (SocketAddress.java:37)
at io.zeebe.broker.system.configuration.SocketBindingCfg.getAddress (SocketBindingCfg.java:26)
at io.zeebe.broker.system.configuration.EmbeddedGatewayCfg.init (EmbeddedGatewayCfg.java:35)
at io.zeebe.broker.system.configuration.BrokerCfg.init (BrokerCfg.java:39)
at io.zeebe.broker.system.configuration.BrokerCfg.init (BrokerCfg.java:29)
at io.zeebe.broker.system.SystemContext.initSystemContext (SystemContext.java:64)
at io.zeebe.broker.system.SystemContext.<init> (SystemContext.java:45)
at io.zeebe.broker.Broker.<init> (Broker.java:92)
at io.zeebe.broker.StandaloneBroker.createBrokerFromConfiguration (StandaloneBroker.java:57)
at io.zeebe.broker.StandaloneBroker.main (StandaloneBroker.java:27)
```

</p>
</details>

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

 Replace the self build `SocketAddress` with the java build in `InetSocketAddress`. The `SocketAddress`  was created to be used in a garbage free env. 
We do not expect to create to much addresses so it gives us no much value and limits the host name to 128 bytes.

Was not able to write a good test where the broker starts with a long host name, since atomix tries to resolve the name then.

## Related issues

closes #3867

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Review by npepinpe — APPROVED** [review]
:+1: 

One possible way would be to write a test with 3 test containers and give them very long hostnames, which should be resolved through the virtual network. But that's kind of a heavyweight test, and it does feel like we're testing the `InetSocketAddress` implementation and not our own code, so I would leave it out.

**Comment by Zelldon** [comment]
Yes I think this is maybe to heavy for such kind of small thing. I think I will add something like that

```

  @Test
  public void shouldNotFailOnLongHostName() {
    // given
    final var port = 1000;
    final var hostName = RandomString.make(139);
    final var networkCfg = new NetworkCfg();
    networkCfg.setPort(port);

    // when
    networkCfg.setHost(hostName);
    final var socketAddress = networkCfg.toSocketAddress();

    // then
    assertThat(socketAddress).isEqualByComparingTo(new SocketAddress(hostName, port));
  }
```

This test fails for example on develop.

**Comment by Zelldon** [comment]
Ok maybe it makes sense to still have the SocketAddress, which just wraps the InetSocketAddress and provides functionallity to get address as `"host:port"` string. There a several places in the code where we use toString of the `SocketAddress` and expect `"host:port"`, but `InetSocketAddress` returns `/host:port`. We would either need to remove the slash or conact host and port with `:`, which is not nice. I probably bring back the `SocketAddress` which then wraps the InetSocketAddress or do you have other ideas @npepinpe

**Comment by npepinpe** [comment]
I would have hoped we don't need anything. Are those usages of `"host:port"` legitimate? As opposed to, z.B. mostly for convenience in tests?

**Comment by Zelldon** [comment]
I have hoped that as well:

 * https://github.com/zeebe-io/zeebe/blob/133c062d6067336f1ece3a2d0ace395ed42d99fe/broker/src/main/java/io/zeebe/broker/system/configuration/EmbeddedGatewayCfg.java#L35
 * https://github.com/zeebe-io/zeebe/blob/fff771a525442f86c966b2f9c7ea47b1eb8df3ba/qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/ClusteringRule.java#L260
 * https://github.com/zeebe-io/zeebe/blob/133c062d6067336f1ece3a2d0ace395ed42d99fe/broker/src/main/java/io/zeebe/broker/Broker.java#L145

**Comment by Zelldon** [comment]
@npepinpe I added in my last commit a helper which converts the host and port to the expected string. As alternative we could again introduce the wrapper class :man_shrugging: you have to decided.

**Review by npepinpe — APPROVED** [review]
:+1: 

I assume I won't need to review after you fix the merge conflicts, but feel free to re-request if you feel you had to change more and want me to take another look

**Comment by Zelldon** [comment]
@npepinpe so helper class is fine?

**Comment by npepinpe** [comment]
I'd rather have neither, but I prefer the helper over a wrapper class to isolate the usages - we can hopefully eventually also get rid of it.

File: util/src/main/java/io/zeebe/util/SocketUtil.java [hunk 1]
```diff
@@ -16,6 +16,6 @@ private SocketUtil() {}
 
   public static String toHostAndPortString(InetSocketAddress inetSocketAddress) {
     return String.format(
-        HOST_PORT_FORMAT, inetSocketAddress.getHostName(), inetSocketAddress.getPort());
+        HOST_PORT_FORMAT, inetSocketAddress.getHostString(), inetSocketAddress.getPort());
```
Comment (npepinpe) [inline util/src/main/java/io/zeebe/util/SocketUtil.java]: :+1: 

Good catch, was unaware of the distinction of these two APIs

**Pull request merged**